update-stardoc-docs.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Renerates the Stardoc rule documentation from source in @io_bazel.
+#
+# This should only need to be run for cutting a new Stardoc release.
+
+set -eu
+
+echo "** Generating Stardoc documentation..."
+bazel build //stardoc:stardoc_doc.md
+
+echo "** Copying result to docs/stardoc_rule.md ..."
+cp bazel-bin/stardoc/stardoc_doc.md docs/stardoc_rule.md
```

Comment:
if i'm not wrong, `bazel-bin/stardoc/stardoc_doc.md` will have its executable bit set. shouldn't be an issue, but just fyi

this is why I did the chmod dance here: https://github.com/bazelbuild/rules_jvm_external/blob/fae429a0c3a5c0da4d4380eeaf347f5331083a5f/scripts/generate_docs.sh#L11

Reply:
By running on my mac, it doesn't seem to have the executable bit set for me, seems set to -rw-r--r--

Maybe genrule behaves differently than stardoc? Genrule might automatically set the executable bit of its `out`, though the stardoc java binary does not.

Reply:
Hm, yeah. good catch.